### PR TITLE
+ configurable endpoints

### PR DESCRIPTION
**Pull request by pape-io**
- Adds configuration ability to each LHS:Record (see added documentation)
- This one allows to use endpoints with different keys for pagination (like search)

### CODE REVIEW

File: lib/lhs/collection.rb [hunk 1]
```diff
@@ -7,17 +7,18 @@ class LHS::Collection < LHS::Proxy
   include InternalCollection
 
   delegate :select, to: :_collection
+  delegate :_record, to: :_data
 
   def total
-    _data._raw[:total]
+    _data._raw[_record.total_key]
```
Comment (kot-begemot) [inline lib/lhs/collection.rb]: Does underscore here stands for private attribute?

File: lib/lhs/concerns/record/endpoints.rb [hunk 2]
```diff
@@ -31,7 +31,7 @@ def endpoint(url, options = nil)
 
       def for_url(url)
         return unless url
-        _template, record = LHS::Record::Endpoints.all.detect do |template, _record_class|
+        _template, record = LHS::Record::Endpoints.all.detect do |template, _record|
```
Comment (kot-begemot) [inline lib/lhs/concerns/record/endpoints.rb]: Does the record is used here anywhere? if not then you can replace it with `_`

**Comment by kot-begemot** [comment]
Two comments, and if rubocop is happy, i am is happy :+1:

File: lib/lhs/collection.rb (hunk 1 above)
Comment (pape-io) [inline lib/lhs/collection.rb]: oui

File: lib/lhs/concerns/record/endpoints.rb (hunk 2 above)
Comment (pape-io) [inline lib/lhs/concerns/record/endpoints.rb]: Good point. Will do.

File: lib/lhs/collection.rb (hunk 1 above)
Comment (pape-io) [inline lib/lhs/collection.rb]: Actually it means 'internal'. So it's not private. It's accessible, but shouldn't be used from consumers of the library.

File: lib/lhs/concerns/record/all.rb [hunk 3]
```diff
@@ -12,12 +12,10 @@ module ClassMethods
       # for the following pages and concatenate all the results in order to return
       # all the objects for a given resource.
       def all(params = {})
-        all = []
-        default_max_limit = 100
-        data = request(params: params.merge(limit: default_max_limit))
-        all.concat(all_items_from(data))
-        request_all_the_rest(data, all, params) if data._raw.is_a?(Hash) && data._raw[:total]
-        data._record_class.new(LHS::Data.new(all, nil, self))
+        limit = params[limit_key] || 100
```
Comment (acoiro) [inline lib/lhs/concerns/record/all.rb]: magic number!

**Comment by acoiro** [comment]
:+1:

File: lib/lhs/collection.rb (hunk 1 above)
Comment (eduardschaeli) [inline lib/lhs/collection.rb]: Not to open a can of worms here, but isn't that what ```protected``` is for?

Comment (pape-io) [inline lib/lhs/collection.rb]: The idea of using some internals is to flag variables that 'could' be used from outside, but can change behaviour at anytime. As the project is getting more and more mature, we can consider moving to protected attributes :+1:.

Comment (pape-io) [inline lib/lhs/collection.rb]: Those `internal` attributes also help exploring and understanding structures.

**Pull request merged**